projects/storefrontlib/src/cms-components/checkout/components/payment-method/payment-method.component.ts

```diff
@@ -83,64 +82,92 @@ export class PaymentMethodComponent implements OnInit, OnDestroy {
       });
 
     this.existingPaymentMethods$ = this.userPaymentService.getPaymentMethods();
-    this.getPaymentDetailsSub = this.checkoutPaymentService
-      .getPaymentDetails()
-      .pipe(
-        filter(
-          (paymentInfo) => paymentInfo && !!Object.keys(paymentInfo).length
-        )
-      )
-      .subscribe((paymentInfo) => {
-        if (this.allowRouting) {
-          this.routingService.go(this.checkoutStepUrlNext);
-        }
-        if (!paymentInfo['hasError']) {
-          this.selectedPayment = paymentInfo;
-        } else {
-          Object.keys(paymentInfo).forEach((key) => {
-            if (key.startsWith('InvalidField')) {
-              this.sendPaymentMethodFailGlobalMessage(paymentInfo[key]);
-            }
-          });
-          this.checkoutService.clearCheckoutStep(3);
-        }
-      });
-  }
 
-  getCardContent(payment: PaymentDetails): Observable<Card> {
-    if (!this.selectedPayment && payment.defaultPayment) {
-      this.selectedPayment = payment;
-    }
+    this.selectedMethod$ = this.checkoutPaymentService.getPaymentDetails().pipe(
+      tap((paymentInfo) => {
+        if (paymentInfo && !!Object.keys(paymentInfo).length) {
+          if (paymentInfo['hasError']) {
+            Object.keys(paymentInfo).forEach((key) => {
+              if (key.startsWith('InvalidField')) {
+                this.sendPaymentMethodFailGlobalMessage(paymentInfo[key]);
+              }
+            });
+            this.checkoutService.clearCheckoutStep(3);
+          } else if (this.shouldRedirect) {
+            this.routingService.go(this.checkoutStepUrlNext);
+          }
+        }
+      })
+    );
 
-    return combineLatest([
-      this.translation.translate('paymentCard.expires', {
-        month: payment.expiryMonth,
-        year: payment.expiryYear,
-      }),
+    this.cards$ = combineLatest([
```

Comment:
I appreciate now this component leverages observables instead of static properties.
At the same time the the `cards$` definition looks a bit too busy. IMO we can refactor it  later w/o breaking change (not MUST for 2.0).

Reply:
Ideally, when card component would handle translation in a different way, it could look much better. Now providing translation with parameters is a pain as you can see.